### PR TITLE
Add exeptions to path safety check

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Adds an exception to the path safety check in the `_get_repo_details` function in `win_pkg`. The exception is `C:\Windows\TEMP`

History:
The `pkg.refresh_db` function deletes the contents of the winrepo directory to avoid having stale files left there that no longer exist on the master. This directory, by default, is `C:\salt\var\cache\salt\minion\files\base\win\repo-ng`. It is feasible, though unlikely, that someone could change the `winrepo_dir` config option to `C:\Windows` in which case `pkg.refresh_db` would begin to delete the contents of `C:\Windows`. The path safety check was added to check for dangerous paths such as `C:\`, `\` and anything in `C:\Windows`.

The Problem:
The test suite places a bunch of files used for testing in the TEMP directory. The test suite is launched from the Jenkins slave using `winexe`. Even though it authenticates as a user on the system, the user profile and environment variables are not loaded. When a user is logged in normally, the temp directory is `C:\Users\<username>\AppData\Local\Temp\`. When logged in via winexe, however, the temp directory is `C:\Windows\TEMP`. This was creating a problem when the `integration.modules.test_pkg` tests tried to run `pkg.refresh_db`, the path to the winrepo, being located in `c:\windows\temp\salt-tests-tmpdir\rootdir\cache\files\base\win\repo-ng`, was failing the safety check because `TEMP` is a folder in `C:\Windows`.

The Fix:
This PR adds an exception for the `C:\Windows\TEMP` directory. This should allow the `pkg.refresh_db` function to complete successfully and the tests to pass.

_NOTE: I tried passing `winexe` the `--profile` switch which is supposed to load the profile, but this had no affect on the `%TEMP%` system variable._

### What issues does this PR fix or reference?
https://github.com/saltstack/zh/issues/1272

### CODE REVIEW

**Comment by twangboy** [comment]
@terminalmage @cachedout Any update on this?

**Comment by cachedout** [comment]
This PR comment that you left is very helpful and quite well-done. Thank you!

This approach seems fine to me but I would like @terminalmage to look this over just as a second pair of eyes before we merge this. I have requested a review from him.

File: salt/modules/win_pkg.py
Comment (terminalmage) [inline salt/modules/win_pkg.py]: I know this exception pre-dates this PR, but is there a way we can improve it? It's not clear what this exception is actually telling the user, and what they would do to remedy it.

**Review by terminalmage — APPROVED** [review]
Looks good, I just had one comment and it's frankly not something that is urgent to take care of in this PR

**Comment by twangboy** [comment]
@terminalmage Changed the error message.

**Pull request merged**
